pkg/storage/segments/store_test.go

```diff
@@ -1,464 +1,196 @@
+			runTest(t, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet, segmentStore segments.Store) {
+				expectedSize := int64(len(tt.data))
+				reader := bytes.NewReader(tt.data)
+
+				beforeModified := time.Now()
+				if tt.err == "" {
+					meta, err := segmentStore.Put(ctx, reader, tt.expiration, func() (storj.Path, []byte, error) {
+						return tt.path, tt.metadata, nil
+					})
+					require.NoError(t, err)
+					assert.Equal(t, expectedSize, meta.Size)
+					assert.Equal(t, tt.metadata, meta.Data)
+					assert.Equal(t, tt.expiration, meta.Expiration)
+					assert.True(t, meta.Modified.After(beforeModified))
+				}
+
+				meta, err := segmentStore.Meta(ctx, tt.path)
+				if tt.err == "" {
+					require.NoError(t, err)
+					assert.Equal(t, expectedSize, meta.Size)
+					assert.Equal(t, tt.metadata, meta.Data)
+					assert.Equal(t, tt.expiration, meta.Expiration)
+					assert.True(t, meta.Modified.After(beforeModified))
+				} else {
+					require.Contains(t, err.Error(), tt.err)
+				}
+			})
 		})
-		assert.NoError(t, err, tt.name)
 	}
 }
 
-func TestSegmentStorePutInline(t *testing.T) {
-	ctrl := gomock.NewController(t)
-	defer ctrl.Finish()
-
+func TestSegmentStorePutGet(t *testing.T) {
 	for _, tt := range []struct {
-		name          string
-		pathInput     string
-		mdInput       []byte
-		thresholdSize int
-		expiration    time.Time
-		readerContent string
+		name       string
+		path       string
+		metadata   []byte
+		expiration time.Time
+		content    []byte
 	}{
-		{"test inline put", "path/1", []byte("111"), 1000, time.Unix(0, 0).UTC(), "readerreaderreader"},
+		{"test inline put/get", "l/path/1", []byte("metadata"), time.Now().UTC(), createTestData(t, 2*memory.KiB.Int64())},
+		{"test remote put/get", "s0/test_bucket/mypath/1", []byte("metadata"), time.Now().UTC(), createTestData(t, 100*memory.KiB.Int64())},
 	} {
-		mockOC := mock_overlay.NewMockClient(ctrl)
-		mockEC := mock_ecclient.NewMockClient(ctrl)
-		mockPDB := mock_pointerdb.NewMockClient(ctrl)
-		mockES := mock_eestream.NewMockErasureScheme(ctrl)
-		rs := eestream.RedundancyStrategy{
-			ErasureScheme: mockES,
-		}
-
-		ss := segmentStore{mockOC, mockEC, mockPDB, rs, tt.thresholdSize}
-		assert.NotNil(t, ss)
-
-		calls := []*gomock.Call{
-			mockPDB.EXPECT().Put(
-				gomock.Any(), gomock.Any(), gomock.Any(),
-			).Return(nil),
-			mockPDB.EXPECT().Get(
-				gomock.Any(), gomock.Any(),
-			),
-		}
-		gomock.InOrder(calls...)
-
-		_, err := ss.Put(ctx, strings.NewReader(tt.readerContent), tt.expiration, func() (storj.Path, []byte, error) {
-			return tt.pathInput, tt.mdInput, nil
+		runTest(t, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet, segmentStore segments.Store) {
+			_, err := segmentStore.Put(ctx, bytes.NewReader(tt.content), tt.expiration, func() (storj.Path, []byte, error) {
+				return tt.path, tt.metadata, nil
+			})
+			require.NoError(t, err, tt.name)
+
+			_, _, err = segmentStore.Get(ctx, tt.path)
```

Comment:
Should we assert that the expected data can be read from the returned ranger? Also that the metadata is the expected one?

Reply:
I will add this